File: Ruby/fizz_buzz.rb
def fizzbuzz(n)
  l = (1..n).map do |i|
    if i % 15 == 0
      'FizzBuzz'
    elsif i % 5 == 0
      'Buzz'
    elsif i % 3 == 0
      'Fizz'
    else
      i
    end
  end
  l.each { |s| p s }
end

fizzbuzz 30

File: Haskell/FizzBuzz.hs
fb :: Int -> IO ()
fb i = mapM_ (print . fb') [1..i]

fb' :: Int -> String
fb' i | i `mod` 15 == 0 = "FizzBuzz"
      | i `mod` 3  == 0 = "Fizz"
      | i `mod` 5  == 0 = "Buzz"
      | otherwise       = (show i)

main :: IO ()
main = fb 30